OSDOCS-13228: container image sig checks MicroShift (#97585)
openshift/openshift-docs: modules/microshift-enable-sigstore-mirror-registries.adoc

// Module included in the following assemblies:
//
// * microshift/microshift_auth_security/microshift-verify-container-signatures.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-enable-sigstore-mirror-registries_{context}"]
= Enabling sigstore attachments for mirror registries

If you are using mirror registries you must apply additional configuration to enable sigstore attachments and mirroring by digest.

.Prerequisites

* You have admin access to the {microshift-short} host.
* You completed the steps in "Verifying container signatures using sigstore."

.Procedure

. Enable sigstore attachments by creating the `/etc/containers/registries.d/mirror.registry.local.yaml` file.
+
[source,terminal]
----
$ cat /etc/containers/registries.d/mirror.registry.local.yaml <1>
docker:
   mirror.registry.local:
        use-sigstore-attachments: true
----

> <1> Optional: Replace `mirror.registry.local.yaml` with your mirror registry URL and filename.

Reviewer: This is not optional. It's most likely that user's registry has another URL.
Let's separate file and URL renames. Please, add <2> on line 24.

```suggestion
<1> Name `mirror.registry.local.yaml` file after your mirror registry URL.
<2> Replace `mirror.registry.local` with your mirror registry URL.
```

> . Enable mirroring by digest by creating the `/etc/containers/registries.conf.d/999-microshift-mirror.conf` file by running the following command:

Reviewer: ```suggestion
. Enable mirroring by digest by creating the `/etc/containers/registries.conf.d/999-microshift-mirror.conf` file by including the following content:
```

> . Enable mirroring by digest by creating the `/etc/containers/registries.conf.d/999-microshift-mirror.conf` file by running the following command and including the Containerfile content:

Reviewer: The command merely lists the contents of the file.
```suggestion
. Enable mirroring by digest by creating the `/etc/containers/registries.conf.d/999-microshift-mirror.conf` with the following contents:
```
